Userland/Utilities/ping.cpp

```diff
@@ -180,6 +180,15 @@ int main(int argc, char** argv)
                 return 1;
             }
 
+            char ihl = -1;
+            memcpy(&ihl, pong_packet.data(), sizeof(char));
+            ihl = ihl & 0x0F;
+            if (!(ihl >= 5) && (ihl <= 15)) {
```

Comment:
The first part of this if is overly complicated:
`!(ihl >= 5)` we can just use `(ihl < 5)`

Also maybe make `5` and `15` into constants as well?
Something like:
`constexpr int min_header_size_in_bytes = 5`

Reply:
that also seems quite weird, `ihl < 5 && ihl <= 15` is just `ihl < 5`.

Reply:
Yes, `ihl` is basically an `internet header length` field in the IPv4 packet. It talks about the length of the complete header in the units of 32-bits word. It has been a `minimum of 5` which is the required part of the header while the optional part can be of `any length subject to a limit of 10 words`. This is an error check to catch possible packet corruption or ill-formed pong packet.